sdk/mediaservices/media-analytics-edge/samples/typescript/src/sampleTemplate.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates something
+ */
+
+import {
+  MediaGraphTopology,
+  MethodRequest,
+  MediaGraphRtspSource,
+  MediaGraphUnsecuredEndpoint,
+  MediaGraphNodeInput,
+  MediaGraphAssetSink,
+  MediaGraphInstance,
+  createMediaGraphTopologySetRequest,
+  createMediaGraphInstanceListRequest,
+  createMediaGraphTopologyListRequest,
+  createMediaGraphTopologyGetRequest,
+  createMediaGraphInstanceSetRequest,
+  createMediaGraphInstanceActivateRequest,
+  createMediaGraphInstanceGetRequest,
+  createMediaGraphInstanceDeActivateRequest,
+  createMediaGraphInstanceDeleteRequest,
+  createMediaGraphTopologyDeleteRequest
+} from "../../../";
+import { Client } from "azure-iothub";
+import { ModuleClient } from "azure-iot-device";
+
+function buildGraphTopology() {
+  const rtspSource: MediaGraphRtspSource = {
+    name: "rtspSource",
+    endpoint: {
+      url: "${rtspUrl}",
+      "@type": "#Microsoft.Media.MediaGraphUnsecuredEndpoint",
+      credentials: {
+        username: "${rtspUserName}",
+        password: "${rtspPassword}",
+        "@type": "#Microsoft.Media.MediaGraphUsernamePasswordCredentials"
+      }
+    } as MediaGraphUnsecuredEndpoint,
+    "@type": "#Microsoft.Media.MediaGraphRtspSource"
+  };
+
+  const graphNodeInput: MediaGraphNodeInput = {
+    nodeName: "rtspSource"
+  };
+
+  const assetSink: MediaGraphAssetSink = {
+    name: "assetSink",
+    inputs: [graphNodeInput],
+    assetNamePattern: "sampleAsset-${System.GraphTopologyName}-${System.GraphInstanceName}",
+    localMediaCachePath: "/var/lib/azuremediaservices/tmp/",
+    localMediaCacheMaximumSizeMiB: "2048",
+    "@type": "#Microsoft.Media.MediaGraphAssetSink"
+  };
+
+  const graphTopology: MediaGraphTopology = {
+    name: "jsTestGraph",
+    properties: {
+      description: "description for jsTestGraph",
+      parameters: [
+        { name: "rtspUserName", type: "String", default: "dummyUsername" },
+        { name: "rtspPassword", type: "SecretString", default: "dumyPassword" },
+        { name: "rtspUrl", type: "String" }
+      ],
+      sources: [rtspSource],
+      sinks: [assetSink]
+    }
+  };
+
+  return graphTopology;
+}
+
+function buildGraphInstance(graphTopologyName: string) {
+  const graphInstance: MediaGraphInstance = {
+    name: graphTopologyName,
+    properties: {
+      description: "description for jsTestGraphInstance",
+      topologyName: "jsTestGraph",
+      parameters: [{ name: "rtspUrl", value: "rtsp://sample.com" }]
+    }
+  };
+
+  return graphInstance;
+}
+
+export async function main() {
+  console.log("== Sample Template ==");
+  const device_id = "lva-sample-device";
+  const module_id = "lvaEdge";
+  const connectionString =
+    "HostName=lvasamplehubwuvvummbzrlk4.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=rHEELfVWEkKxbMBvdc7Fcp49QdBDA6skuOcLIIryD6c=";
```

Comment:
is this a real secret? You should invalidate it if so. I'd recommend using environment variables to avoid committing such things